--- cmake-3.27.3.orig/Source/Checks/cm_cxx_features.cmake	2023-08-16 19:46:35.000000000 +0200
+++ cmake-3.27.3/Source/Checks/cm_cxx_features.cmake	2023-08-23 23:10:16.079757726 +0200
@@ -96,7 +96,7 @@
   set(CMake_HAVE_CXX_FILESYSTEM FALSE)
 endif()
 
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND CMAKE_SYSTEM_PROCESSOR MATCHES "armv7l|sparc")
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
   cm_check_cxx_feature(atomic_builtin)
   if(NOT CMake_HAVE_CXX_ATOMIC_BUILTIN)
     set(cm_check_cxx_feature_LINK_LIBRARIES atomic)
