--- cmake-3.8.0/CMakeLists.txt.orig	2017-04-10 17:23:06.000000000 +0200
+++ cmake-3.8.0/CMakeLists.txt	2017-04-26 22:06:31.968941358 +0200
@@ -539,13 +539,15 @@
   #---------------------------------------------------------------------
   # Build XMLRPC library for CMake and CTest.
   if(CTEST_USE_XMLRPC)
+    find_package(XMLRPC QUIET REQUIRED)
+    set(XMLRPC_BASE_LIBRARIES ${XMLRPC_BASE_LIBRARIES})
     find_package(XMLRPC QUIET REQUIRED libwww-client)
     if(NOT XMLRPC_FOUND)
       message(FATAL_ERROR
         "CTEST_USE_XMLRPC is ON but xmlrpc is not found!")
     endif()
     set(CMAKE_XMLRPC_INCLUDES ${XMLRPC_INCLUDE_DIRS})
-    set(CMAKE_XMLRPC_LIBRARIES ${XMLRPC_LIBRARIES})
+    set(CMAKE_XMLRPC_LIBRARIES ${XMLRPC_LIBRARIES} ${XMLRPC_BASE_LIBRARIES})
   endif()
 
   #---------------------------------------------------------------------
