diff -burN cmake-1.8.3.orig/Source/CursesDialog/ccmake.cxx cmake-1.8.3/Source/CursesDialog/ccmake.cxx
--- cmake-1.8.3.orig/Source/CursesDialog/ccmake.cxx	2004-06-29 23:56:20.839557744 +0200
+++ cmake-1.8.3/Source/CursesDialog/ccmake.cxx	2004-06-29 23:57:35.194254112 +0200
@@ -24,7 +24,7 @@
 
 #include "cmCursesMainForm.h"
 
-#include <curses.h>
+#include <ncurses/curses.h>
 #include <form.h>
 
 //----------------------------------------------------------------------------
diff -burN cmake-1.8.3.orig/Source/CursesDialog/cmCursesStandardIncludes.h cmake-1.8.3/Source/CursesDialog/cmCursesStandardIncludes.h
--- cmake-1.8.3.orig/Source/CursesDialog/cmCursesStandardIncludes.h	2004-06-29 23:56:20.842557288 +0200
+++ cmake-1.8.3/Source/CursesDialog/cmCursesStandardIncludes.h	2004-06-29 23:57:52.848570248 +0200
@@ -28,7 +28,7 @@
 # include <form.h>
 # undef _XOPEN_SOURCE_EXTENDED
 #else
-# include <curses.h>
+# include <ncurses/curses.h>
 # include <form.h>
 #endif
 
diff -burN cmake-1.8.3.orig/Source/CursesDialog/form/form.h cmake-1.8.3/Source/CursesDialog/form/form.h
--- cmake-1.8.3.orig/Source/CursesDialog/form/form.h	2004-06-29 23:56:20.823560176 +0200
+++ cmake-1.8.3/Source/CursesDialog/form/form.h	2004-06-29 23:57:07.869408120 +0200
@@ -37,7 +37,7 @@
  #define _MSE_INT_H
 #endif
 
-#include <curses.h>
+#include <ncurses/curses.h>
 #include <eti.h>
 #include <stdarg.h>
 
