--- cmake-3.27.3/Modules/FindZLIB.cmake.orig	2023-08-16 19:46:35.000000000 +0200
+++ cmake-3.27.3/Modules/FindZLIB.cmake	2023-08-23 16:25:36.123237881 +0200
@@ -179,9 +179,15 @@ if(ZLIB_INCLUDE_DIR AND EXISTS "${ZLIB_I
 
     string(REGEX REPLACE "^.*ZLIB_VERSION \"([0-9]+).*$" "\\1" ZLIB_VERSION_MAJOR "${ZLIB_H}")
     string(REGEX REPLACE "^.*ZLIB_VERSION \"[0-9]+\\.([0-9]+).*$" "\\1" ZLIB_VERSION_MINOR  "${ZLIB_H}")
-    string(REGEX REPLACE "^.*ZLIB_VERSION \"[0-9]+\\.[0-9]+\\.([0-9]+).*$" "\\1" ZLIB_VERSION_PATCH "${ZLIB_H}")
-    set(ZLIB_VERSION_STRING "${ZLIB_VERSION_MAJOR}.${ZLIB_VERSION_MINOR}.${ZLIB_VERSION_PATCH}")
+    set(ZLIB_VERSION_STRING "${ZLIB_VERSION_MAJOR}.${ZLIB_VERSION_MINOR}")
 
+    # only append a PATCH version if it exists:
+    set(ZLIB_VERSION_PATCH "")
+    if( "${ZLIB_H}" MATCHES "ZLIB_VERSION \"[0-9]+\\.[0-9]+\\.([0-9]+)")
+	    set(ZLIB_VERSION_PATCH "${CMAKE_MATCH_1}")
+	    string(APPEND ZLIB_VERSION_STRING ".${ZLIB_VERSION_PATCH}")
+    endif()
+        
     # only append a TWEAK version if it exists:
     set(ZLIB_VERSION_TWEAK "")
     if( "${ZLIB_H}" MATCHES "ZLIB_VERSION \"[0-9]+\\.[0-9]+\\.[0-9]+\\.([0-9]+)")
