diff -urNp -x '*.orig' cmake-3.17.2.org/Auxiliary/CMakeLists.txt cmake-3.17.2/Auxiliary/CMakeLists.txt
--- cmake-3.17.2.org/Auxiliary/CMakeLists.txt	2020-04-28 16:23:06.000000000 +0200
+++ cmake-3.17.2/Auxiliary/CMakeLists.txt	2021-04-12 22:18:36.892988114 +0200
@@ -1,4 +1,4 @@
 install(DIRECTORY vim/indent vim/syntax DESTINATION ${CMAKE_DATA_DIR}/editors/vim)
-install(FILES cmake-mode.el DESTINATION ${CMAKE_DATA_DIR}/editors/emacs)
+install(FILES cmake-mode.el DESTINATION ${CMAKE_INSTALL_DATADIR}/emacs/site-lisp)
 install(FILES cmake.m4 DESTINATION ${CMAKE_XDGDATA_DIR}/aclocal)
 add_subdirectory (bash-completion)
diff -urNp -x '*.orig' cmake-3.17.2.org/Auxiliary/bash-completion/CMakeLists.txt cmake-3.17.2/Auxiliary/bash-completion/CMakeLists.txt
--- cmake-3.17.2.org/Auxiliary/bash-completion/CMakeLists.txt	2020-04-28 16:23:06.000000000 +0200
+++ cmake-3.17.2/Auxiliary/bash-completion/CMakeLists.txt	2021-04-12 22:18:36.892988114 +0200
@@ -5,4 +5,4 @@
 # add symlinks to the files in appropriate places
 #  /etc/bash_completion.d/
 #  DATADIR/completions (may be /usr/share/<package>/completions
-install(FILES cmake cpack ctest DESTINATION ${CMAKE_DATA_DIR}/completions)
+install(FILES cmake cpack ctest DESTINATION ${CMAKE_INSTALL_DATADIR}/bash-completion/completions)
