--- cmake-2.8.12.1/Tests/CMakeLists.txt.orig	2013-11-05 20:07:23.000000000 +0100
+++ cmake-2.8.12.1/Tests/CMakeLists.txt	2013-11-08 15:52:51.604058079 +0100
@@ -2216,6 +2216,7 @@
   set(regex "(Problems when submitting via S*CP")
   set(regex "${regex}|Error message was: ")
   set(regex "${regex}([Cc]ould *n.t resolve host")
+  set(regex "${regex}|[Cc]ould *n.t resolve:")
   set(regex "${regex}|[Cc]ould *n.t connect to host")
   set(regex "${regex}|Empty reply from server")
   set(regex "${regex}|The requested URL returned error")
