--- cmake-3.31.7/Source/cmExportInstallCMakeConfigGenerator.cxx.orig	2025-04-10 15:13:20.000000000 +0200
+++ cmake-3.31.7/Source/cmExportInstallCMakeConfigGenerator.cxx	2025-06-12 20:14:54.901643462 +0200
@@ -123,7 +123,7 @@ bool cmExportInstallCMakeConfigGenerator
   }
 
   this->CleanupTemporaryVariables(os);
-  this->GenerateImportedFileCheckLoop(os);
+  // disabled for distro packages: this->GenerateImportedFileCheckLoop(os);
 
   // Generate an import file for each configuration.
   // Don't do this if we only export INTERFACE_LIBRARY targets.
